docs/aws/audit/elbmonitoring/rules/elb_connection_draining_remediation.mdx

```diff
@@ -1,6 +1,221 @@
+
+By following these steps, you ensure that connection draining is enabled for your ELB, which helps in gracefully terminating existing connections during instance deregistration or shutdown.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the misconfiguration where Elastic Load Balancers (ELBs) should have connection draining enabled using AWS CLI, follow these steps:
+
+1. **Describe the Load Balancer:**
+   First, identify the load balancer for which you want to enable connection draining.
+   ```sh
+   aws elb describe-load-balancers --load-balancer-names <your-load-balancer-name>
+   ```
+
+2. **Enable Connection Draining:**
+   Use the `modify-load-balancer-attributes` command to enable connection draining. You need to specify the load balancer name and the connection draining attributes.
+   ```sh
+   aws elb modify-load-balancer-attributes --load-balancer-name <your-load-balancer-name> --load-balancer-attributes '{"ConnectionDraining": {"Enabled": true, "Timeout": 300}}'
+   ```
+
+3. **Verify the Configuration:**
+   Confirm that connection draining has been enabled by describing the load balancer attributes.
+   ```sh
+   aws elb describe-load-balancer-attributes --load-balancer-name <your-load-balancer-name>
+   ```
+
+4. **Automate the Process:**
+   Optionally, you can create a script to automate this process for multiple load balancers. Here is a simple example in bash:
+   ```sh
+   for lb in $(aws elb describe-load-balancers --query 'LoadBalancerDescriptions[*].LoadBalancerName' --output text); do
+       aws elb modify-load-balancer-attributes --load-balancer-name $lb --load-balancer-attributes '{"ConnectionDraining": {"Enabled": true, "Timeout": 300}}'
+   done
+   ```
+
+These steps will help ensure that connection draining is enabled for your ELBs, thereby preventing the misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration where Elastic Load Balancers (ELBs) should have connection draining enabled in AWS, you can use the Boto3 library in Python. Connection draining ensures that the load balancer stops sending requests to instances that are deregistering or unhealthy, allowing existing requests to complete.
+
+Here are the steps to ensure connection draining is enabled using a Python script:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+3. **Python Script to Enable Connection Draining**:
+   Use the following Python script to enable connection draining for your ELBs.
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EC2
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>